node-repository/src/main/java/com/yahoo/vespa/hosted/provision/lb/SharedLoadBalancerService.java

```diff
@@ -0,0 +1,56 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.provision.lb;
+
+import com.yahoo.config.provision.ApplicationId;
+import com.yahoo.config.provision.ClusterSpec;
+import com.yahoo.config.provision.HostName;
+import com.yahoo.config.provision.NodeType;
+import com.yahoo.vespa.hosted.provision.NodeRepository;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * @author ogronnesby
+ */
+public class SharedLoadBalancerService implements LoadBalancerService {
+    private final NodeRepository nodeRepository;
+
+    public SharedLoadBalancerService(NodeRepository nodeRepository) {
+        this.nodeRepository = Objects.requireNonNull(nodeRepository, "Missing nodeRepository value");
+    }
+
+    @Override
+    public LoadBalancerInstance create(ApplicationId application, ClusterSpec.Id cluster, Set<Real> reals) {
+        final var proxyNodes = nodeRepository.getNodes(NodeType.proxy);
+
+        if (proxyNodes.size() == 0) {
+            throw new RuntimeException("Missing proxy nodes in nodeRepository");
```

Comment:
Consider `IllegalStateException`. This can only happen if an application is deployed *before* any proxy nodes are provisioned, i.e. an illegal state. 🙂

Reply:
Fixed